protocol/metrics/consumer_optimizer_qos_client.go

```diff
@@ -200,96 +189,88 @@ func (coqc *ConsumerOptimizerQoSClient) RegisterOptimizer(optimizer OptimizerInf
 		return
 	}
 
-	coqc.optimizers.Store(chainId, optimizer)
-}
+	coqc.lock.Lock()
+	defer coqc.lock.Unlock()
 
-func (coqc *ConsumerOptimizerQoSClient) initAtomicUint64() *atomic.Uint64 {
-	newAtomic := &atomic.Uint64{}
-	newAtomic.Add(1)
-	return newAtomic
+	if _, found := coqc.optimizers[chainId]; found {
+		utils.LavaFormatWarning("Optimizer already registered for chain", nil, utils.LogAttr("chainId", chainId))
+		return
+	}
+
+	coqc.optimizers[chainId] = optimizer
 }
 
-func (coqc *ConsumerOptimizerQoSClient) incrementStoreCounter(store *common.SafeSyncMap[string, *common.SafeSyncMap[string, *atomic.Uint64]], providerAddress, chainId string) {
+func (coqc *ConsumerOptimizerQoSClient) incrementStoreCounter(store map[string]map[string]uint64, chainId, providerAddress string) {
```

Comment:
passing a store by value and not by pointer. changes on it doesnt affect the original store

Reply:
Talked about this offline, resolving